File: mobile-engage/src/main/java/com/emarsys/mobileengage/deeplink/DeepLinkInternal.java
package com.emarsys.mobileengage.deeplink;

import android.app.Activity;
import android.content.Intent;

import com.emarsys.core.api.result.CompletionListener;

public interface DeepLinkInternal {
    void trackDeepLinkOpen(Activity activity, Intent intent, CompletionListener completionListener);
}

File: mobile-engage/src/main/java/com/emarsys/mobileengage/deeplink/DeepLinkInternal.kt
package com.emarsys.mobileengage.deeplink

import android.app.Activity
import android.content.Intent
import com.emarsys.core.api.result.CompletionListener

interface DeepLinkInternal {
    fun trackDeepLinkOpen(
        activity: Activity,
        intent: Intent,
        completionListener: CompletionListener?
    )
}